src/sp/it/pl/util/async/future/Fut.kt

```diff
@@ -0,0 +1,150 @@
+package sp.it.pl.util.async.future
+
+import javafx.scene.control.ProgressIndicator
+import javafx.util.Duration
+import mu.KLogging
+import sp.it.pl.main.APP
+import sp.it.pl.util.async.FX
+import sp.it.pl.util.async.NEW
+import sp.it.pl.util.functional.kt
+import sp.it.pl.util.async.sleep
+import sp.it.pl.util.functional.Try
+import sp.it.pl.util.functional.invoke
+import sp.it.pl.util.functional.toUnit
+import java.util.Optional
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.Executor
+import java.util.function.Consumer
+
+/**
+ * Future monad implementation.
+ *
+ * Oriented for practicality * ease of use, not specification (monadic laws) or robustness (API completeness).
+ */
+class Fut<T>(private var f: CompletableFuture<T>) {
+
+    fun cancel(): Unit = f.cancel(true).toUnit()
```

Comment:
`.toUnit()`
I know what this is for, but it seems so wrong. Why not simply return whatever the underlying cancel function returns?

Reply:
Yes I was expecting a feedback.

Because that would be leaking implementation details.

Ultimately toUnit() makes no sense anyway, Ill use an apply {} block instead. I do not know what I was thinking. Cool experiment though and it works. Kinda puts the type system into new perspective.

Reply:
You could also just create a one-lined function with braces, aka `fun cancel() { f.cancel(true) }`
There is a code style setting in IDEA explicitly for leaving one-lined functions on a single line.

Reply:
I'd rather reserve that syntax for lambda only

Reply:
It is harmless, lets see what toUnit() can do. Experiment.